cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md

```diff
@@ -901,6 +901,42 @@ void trivial_types_only(T t){
 }
 ```
 
+# Logging
+
+libcudf includes logging utilities (built on top of [spdlog](https://github.com/gabime/spdlog)
+library), which should be used to log important events (e.g. user warnings). This utility can also
+be used to log debug information, as long as the correct logging level is used. There are six macros
+that should be used for logging at different levels:
+
+* `CUDF_LOG_TRACE` - verbose debug messages (targeted at developers)
+* `CUDF_LOG_DEBUG` - debug messages (targeted at developers)
+* `CUDF_LOG_INFO` - information about rare events (e.g. once per run) that occur during normal
+execution
+* `CUDF_LOG_WARN` - user warnings about potentially unexpected behavior or deprecations
+* `CUDF_LOG_ERROR` - recoverable errors
+* `CUDF_LOG_CRITICAL` - unrecoverable errors (e.g. memory corruption)
+
+By default, messages below the `INFO` level are excluded from the log. In public builds, the code
```

Comment:
What does "below" mean? Are they "WARN"+"ERROR"+"CRITICAL" (as they are "below" "INFO" in the list above)? They seem important and should not be "compiled out in public build".

Reply:
below here refers to urgency/importance. Will rephrase to remove ambiguity.